src/main/java/frc/robot/commands/autonomousCommands/AutonomousShooterCommand.java

```diff
@@ -26,24 +27,28 @@
     private HopperController hopperController;
     private IndexerController indexerController;
     private DistanceSensorDetector distanceSensor;
+
+    //Commands
+    private IndexerIntakeCommand indexerIntakeCommand;
+
     //variables
     private double ballsShot = 0;
-    private boolean ballShot = false;
-    private double startTime;
-    private boolean ballDetected = false;
+    private double startTime = 0;
+    private boolean ballinIndexer = false;
     private double distanceToGoal = 149.0; 
 
+
     //constants
     private final double INNER_DISTANCE_FROM_TARGET = 29.0;
     private final double LAUNCH_TIME = 0.5;
     private final double AUTONOMOUS_SPEED = .6;
 
     public AutonomousShooterCommand(Drive drive, ShooterSpeedController shooterSpeedController,
-    HopperController hopperController, IndexerController indexerController){
+    HopperController hopperController, IndexerIntakeCommand indexerIntakeCommand){
```

Comment:
You had this right the first time, you want to use the Controller.  Commands call subsystems, not other commands.

Reply:
Command Groups and the Operator Inputs class will call Commands.

For the most part, we will want to run one command at a time.  The ArcadeDrive/CheesyDrive command is an exception, because it's always running during Teleop, and then other commands also get scheduled.  For those other commands, we will only do one at a time.  There is one subsystem, drive that is used by all sorts of commands and we only want one at a time actually using it, which is why we have addRequirements(drive) all over the place, but we don't really need it for any other subsystems.  I apologize if you already know all this.